internal/planner/planner.go

```diff
@@ -299,15 +300,24 @@ func (p *Planner) planRules(rules []*ast.Rule) (string, error) {
 	// Default rules execute if the return is undefined.
 	if defaultRule != nil {
 
-		fn.Blocks = append(fn.Blocks, &ir.Block{
-			Stmts: []ir.Stmt{
-				&ir.IsUndefinedStmt{Source: fn.Return},
-			},
-		})
+		// Set the location for the default rule head.
+		p.loc = defaultRule.Loc()
+		// NOTE(sr) for `default p = 1`,
+		// defaultRule.Loc() is `default`,
+		// defaultRule.Head.Loc() is `p = 1`.
+		// I'd prefer we'd have `default p = 1`; but since the mapped
+		// locations are col/row only (no extent), pointing to `default`
+		// seems preferable.
+
+		fn.Blocks = append(fn.Blocks, p.blockWithStmt(&ir.IsUndefinedStmt{Source: fn.Return}))
+
+		p.loc = defaultRule.Head.Loc()
 
 		p.curr = fn.Blocks[len(fn.Blocks)-1]
 
+		// TODO(sr): default rules can have bodies?!
```

Comment:
For better or worse, they're just represented as normal rules. The parser will generate a body that never fails (`true`). This could be useful in the future we if we wanted to relax some of the restrictions around `default`...

```
package x

default p = q   # q is always defined so this is fine however today it'll be rejected

default q = 7

Reply:
💡 Ah, I see. Thanks!